Solutions/Trend Micro Vision One/Data Connectors/AzureFunctionTrendMicroXDR/shared_code/utils.py

```diff
@@ -27,6 +27,16 @@ def find_token_by_clp(clp_id, api_tokens):
     return next(filter(lambda token: get_clp_id(token) == clp_id, api_tokens), None)
 
 
+def check_token_is_expired(token: str) -> bool:
+    try:
+        return datetime.now() > datetime.fromtimestamp(
+            jwt.decode(token, options={"verify_signature": False}).get('et')
+        )
+    except Exception as e:
+        logger.error(f"if_token_is_expired checking Error, e: {e}")
+        return False
```

Comment:
@V1ManagedServices , the return result will be False, hope that is correct, thanks.

Reply:
hi @v-sabiraj, it is correct. This is a temporary enhancement to prevent the use of some expired tokens. If there are any tokens not in the desired format, we'd like to process them using the old flow. Thanks for pointing this out.